### PR TITLE
Carve a solana-program-interface crate out of solana-sdk

### PR DESCRIPTION
**Pull request by jackcmay**
#### Summary of Changes

This is the first cut to `solana-program-interface` crate and does not yet touch `solana-sdk`.  The token and memo programs have been verified to build against `solana-program-interface`.
 

Fixes #

### CODE REVIEW

**Comment by jackcmay** [comment]
fyi @mvines @CriesofCarrots

File: Cargo.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ members = [
     "metrics",
     "net-shaper",
     "notifier",
+    "program-interface",
```
Comment (mvines) [inline Cargo.toml:34]: I think `program-interface` won't live in here.  It would be a stand-alone crate with it's own `[workspace]`

Comment (jackcmay) [reply Cargo.toml:34]: Yeah, that is temporary, everything outside of `/program-interface` is temp

File: program-interface/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[package]
+name = "solana-program-interface"
```
Comment (garious) [inline program-interface/Cargo.toml:2]: I'm confused about the name and scope of this crate.  That `-interface` suffix suggests it'll only be a dependency of `instruction` and `state` modules, and that `processor` modules might require an additional dependency. Is that correct? If so, why is `entrypoint.rs` here? If no, can we rename to `solana-program`?

**Comment by jackcmay** [comment]
@garious This was a trial run and intended to be the bare minimum needed by programs (hence entrypoint, etc...).  I threw it up here so that @mvines and @CriesofCarrots could get a feeling of what that might look like while we were discussing the various options.  As far as naming we would shurly run it accross your desk before being commited :-)

**Pull request closed without merging**